tests/ui/non_canonical_clone_impl.fixed

```diff
@@ -95,3 +96,7 @@ impl<A: Copy> Clone for Uwu<A> {
 }
 
 impl<A: std::fmt::Debug + Copy + Clone> Copy for Uwu<A> {}
+
+// should skip proc macros, see https://github.com/rust-lang/rust-clippy/issues/12788
+#[derive(proc_macro_derive::NonCanonicalClone)]
+pub struct G;
```

Comment:
While we're here, please also add a test case using `proc_macros::with_span` [like here](https://github.com/rust-lang/rust-clippy/blob/master/tests/ui/as_conversions.rs#L17). That's another common FP case.

Might need to add another check using `is_from_proc_macro` to the lint.

Reply:
Sorry, what I meant was something like like
```rs
with_span!(
    span

    #[derive(Copy)]
    struct H;
    impl Clone for H {
        fn clone(&self) -> Self {
            todo!()
        }
    }
);
```
(not using the proc macro) ^^

Reply:
Sorry, it's my fault.
Thank you so much.